src/components/controllers/linked-block-controller.ts

```diff
@@ -0,0 +1,106 @@
+import { ReactiveController } from 'lit';
+import { OutlineElement } from '../base/outline-element/outline-element';
+
+interface OutlineElementWithHref extends OutlineElement {
+  href?: string;
+}
+
+/**
+ * Support accessible linked blocks.
+ *
+ * See https://inclusive-components.design/cards/.
+ *
+ * Blocks can function as a single clickable object while supporting good screen reader features.
+ * - The link is focusable and shows in the accessibility tree with a short description.
+ * - Each of the pieces of the component are available individually if not cosmetic.
+ *
+ * Supports click or drag to select text.
+ */
+export class LinkedBlockController implements ReactiveController {
+  host: OutlineElementWithHref;
+
+  // This needs to be a method so that we can target elements that don't exist until the component is done rendering.
+  private _getLink: () => HTMLElement | undefined;
+
+  // Try and support clicking and selecting text.
+  private _lastMouseDownTimestamp = 0;
+
+  constructor(
+    host: OutlineElementWithHref,
+    getLink: () => HTMLElement | undefined
+  ) {
+    this.host = host;
+    this.host.addController(this);
+
+    this._getLink = getLink;
+
+    // Bind our methods so that `this` doesn't change depending on how the methods are called.
+    // See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this#this_in_classes.
+    this.isLinked = this.isLinked.bind(this);
+  }
+
+  hostConnected() {
+    if (this.isLinked()) {
+      this._handleClicks();
+      this._handleHover();
+      this._handleFocus();
+    }
+  }
+
+  isLinked(): boolean {
+    return this.host.href !== undefined && this.host.href.length > 0;
```

Comment:
Want to point out that this uses the pattern, if component has `href` as an attribute, it will function as a link.

I believe we decided we didn't want to use that pattern of using attribute names that already exist.

I think @sarahp updated this for a project.